compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -1150,14 +1163,28 @@ class Namer { typer: Typer =>
                 case tp: TermRef => tp.termSymbol.is(Private) || refersToPrivate(tp.prefix)
                 case _ => false
               val (maybeStable, mbrInfo) =
-                if sym.isStableMember && sym.isPublic && !refersToPrivate(path.tpe) then
-                  (StableRealizable, ExprType(path.tpe.select(sym)))
+                if sym.isStableMember
+                    && sym.isPublic
+                    && pathType.isStable
+                    && !refersToPrivate(pathType)
+                then
+                  (StableRealizable, ExprType(pathType.select(sym)))
                 else
-                  (EmptyFlags, mbr.info.ensureMethodic)
+                  def addPathMethodParams(pt: Type, info: Type): Type = pt match
```

Comment:
this does not check for parameter name conflicts, e.g.
```scala
class StrOps(x: String):
  def mergeWith(y: String): String = x + y

extension (y: String) {
  private def ops = new StrOps(y)
  export ops.mergeWith
}
```

produces the method:
```scala
extension (y: String) final def mergeWith(y: String): String = 
  app.app$package.ops(y).mergeWith(y)
```

Reply:
OK, let's try to rename parameters if they clash.